src/ty/mod.rs

```diff
@@ -205,8 +218,14 @@ where
     T: Form,
 {
     /// Create a new [`TypeParameter`].
-    pub fn new(name: T::String, ty: Option<T::Type>) -> Self {
-        Self { name, ty }
+    pub fn new<S>(name: S, ty: Option<T::Type>) -> Self
+    where
+        S: Into<T::String>,
```

Comment:
This is very silly/edge casey but it's possible for this to be a breaking change because the type checker doesn't know quite so much about what `name` could be any more. Stupid contrived example:

```rust
fn into_string(val: impl Into<String>) {}
fn concrete_string(val: String) {}

fn main() {
    let maybe_string = ['a','b','c'].into_iter().collect();
    // breaks if changed to into_string:
    concrete_string(maybe_string);
}
```

Reply:
Possibly, but I have split this into `new` (legacy for MetaForm) and added a new `new_portable` method as I have for other types here. https://github.com/paritytech/scale-info/pull/164/commits/c9149c2e88812af32d6a85b9edcdc97c2f837652